### ISSUE TITLE
Generating .so file results in link error

### ISSUE THREAD

**Issue by arlencox** [ocaml/dune#1310]
Change #1172 seems to have broken my build of code that generates a linux .so file.  There appears to be a problem with how the code that is generated due to #1172 is compiled.

test.ml:
```
let _ =
  Findlib.init ()
```
dune:
```
(executable
  (name test)
   (modes (native shared_object))
   (libraries findlib findlib.dynload)
  )
```
Build command:
```
$ dune build test.so
    ocamlopt test.so (exit 2)
(cd _build/default && ocamlopt.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -o test.so -cclib -lm -cclib -ldl -output-complete-obj -runtime-variant _pic -linkall -I opam/4.07.0/lib/findlib opam/4.07.0/lib/findlib/findlib.cmxa opam/4.07.0/lib/ocaml/dynlink.cmxa opam/4.07.0/lib/findlib/findlib_dynload.cmxa .test.eobjs/test_findlib_initl_native.cmx .test.eobjs/test.cmx)
ld: .test.eobjs/test_findlib_initl_native.o: relocation R_X86_64_PC32 against symbol `camlFindlib' can not be used when making a shared object; recompile with -fPIC
ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
File "caml_startup", line 1:
Error: Error during linking
```

Build log:
```
# dune build test.so
# OCAMLPARAM: unset
$ /bin/nproc > /tmp/dune1a0ff9.output 2> /dev/null
$ opam/4.07.0/bin/ocamlc.opt -config > /tmp/dune8d0bd3.output
$ (cd _build/default && opam/4.07.0/bin/ocamldep.opt -modules -impl test.ml) > _build/default/.test.eobjs/test.ml.d
$ (cd _build/default && opam/4.07.0/bin/ocamlc.opt -bin-annot -I .test.eobjs -I opam/4.07.0/lib/findlib -no-alias-deps -opaque -o .test.eobjs/test_findlib_initl_native.cmo -c -impl .test.eobjs/test_findlib_initl_native.ml)
$ (cd _build/default && opam/4.07.0/bin/ocamlc.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -bin-annot -I .test.eobjs -I opam/4.07.0/lib/findlib -no-alias-deps -opaque -o .test.eobjs/test.cmo -c -impl test.ml)
$ (cd _build/default && opam/4.07.0/bin/ocamlopt.opt -I .test.eobjs -I opam/4.07.0/lib/findlib -intf-suffix .ml -nodynlink -no-alias-deps -opaque -o .test.eobjs/test_findlib_initl_native.cmx -c -impl .test.eobjs/test_findlib_initl_native.ml)
$ (cd _build/default && opam/4.07.0/bin/ocamlopt.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -I .test.eobjs -I opam/4.07.0/lib/findlib -intf-suffix .ml -no-alias-deps -opaque -o .test.eobjs/test.cmx -c -impl test.ml)
$ (cd _build/default && opam/4.07.0/bin/ocamlopt.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -o test.so -cclib -lm -cclib -ldl -output-complete-obj -runtime-variant _pic -linkall -I opam/4.07.0/lib/findlib opam/4.07.0/lib/findlib/findlib.cmxa opam/4.07.0/lib/ocaml/dynlink.cmxa opam/4.07.0/lib/findlib/findlib_dynload.cmxa .test.eobjs/test_findlib_initl_native.cmx .test.eobjs/test.cmx)
> ld: .test.eobjs/test_findlib_initl_native.o: relocation R_X86_64_PC32 against symbol `camlFindlib' can not be used when making a shared object; recompile with -fPIC
> ld: final link failed: Bad value
> collect2: error: ld returned 1 exit status
> File "caml_startup", line 1:
> Error: Error during linking
[2]
```
(note: paths have been sanitized to remove user names)

**Comment by diml** [comment]
@bobot looking at the compilation commands, it's because we are passing `-nodynlink` when compiling the generated file for findlib. This is because of passing `~dynlink:false` to `Module_compilation.build_module` in `link_time_code_gen.ml`. Remove this should fix the issue. `~dynlink:false` was meant for handling `(no_dynlink)`.

**Comment by diml** [comment]
Fixed in #1317

**Comment by arlencox** [comment]
I can confirm that 1.3.0 fixes the problem on my machine.

Thanks.
